test.js

```diff
@@ -133,3 +133,12 @@ test('PCancelable.fn()', async t => {
 test('PCancelable.CancelError', t => {
 	t.true(PCancelable.CancelError.prototype instanceof Error);
 });
+
+test.failing('resolves on cancel', async t => {
+	const p = new PCancelable((onCancel, resolve, reject) => {
+		onCancel(() => {});
+	});
+	setTimeout(() => { p.cancel(); }, 100);
+	await t.throws(Promise.resolve(p));
+	await t.throws(p);
```

Comment:
Can you explain this? How come it's settled? I don't resolve it at any point from what I can tell.